scripts/make-g-cloud-7-live.py

```diff
@@ -0,0 +1,120 @@
+"""
+
+Usage:
+    scripts/make-g-cloud-7-live.py <data_api_url> <data_api_token> <draft_bucket> <documents_bucket> [--dry-run]
+"""
+import sys
+sys.path.insert(0, '.')
+
+from itertools import islice
+import random
+import re
+
+from docopt import docopt
+from dmutils.apiclient import DataAPIClient
+from dmutils.s3 import S3
+
+DOCUMENT_KEYS = [
+    'pricingDocumentURL', 'serviceDefinitionDocumentURL',
+    'sfiaRateDocumentURL', 'termsAndConditionsDocumentURL',
+]
+FRAMEWORK_SLUG = 'g-cloud-7'
+
+
+def assert_equal(one, two):
+    if one != two:
+        raise AssertionError("{} != {}".format(one, two))
+
+
+def find_suppliers_on_framework(client, framework_slug):
+    return (
+        supplier for supplier in client.find_framework_suppliers(FRAMEWORK_SLUG)['supplierFrameworks']
+        if supplier['onFramework']
+    )
+
+
+def find_submitted_draft_services(client, supplier, framework_slug):
+    return (
+        draft_service for draft_service in
+        client.find_draft_services(supplier['supplierId'], framework=framework_slug)['services']
+        if draft_service['status'] == 'submitted' and not supplier.get('serviceId')
+    )
+
+
+def parse_document_url(url, framework_slug):
+    pattern = r'/{}/(\d+)/(\d+)-(.*)$'.format(re.escape(framework_slug))
+    match = re.search(pattern, url)
+    if not match:
+        raise ValueError("Could not parse document URL {}".format(url))
+    return {
+        'supplier_id': match.group(1),
+        'draft_id': match.group(2),
+        'document_name': match.group(3),
+    }
+
+
+def get_draft_document_path(parsed_document, framework_slug):
+    return '{framework_slug}/{supplier_id}/{draft_id}-{document_name}'.format(
+        framework_slug=framework_slug,
+        **parsed_document)
+
+
+def get_live_document_path(parsed_document, framework_slug, service_id):
+    return '{framework_slug}/documents/{supplier_id}/{service_id}-{document_name}'.format(
```

Comment:
@TheDoubleK `/documents/` in the middle is what we've agreed on during the URL discussions on that Google doc. This makes the paths more consistent between multiple buckets and makes the nginx dispatch rules simpler for the new URLs. We'll need to special case the old-style asset URLs though, but that's something we're aware of.